### PR TITLE
SAML Health Check

### PR DESCRIPTION
**Pull request by omgitsbillryan**
**Related Issue**
https://github.com/department-of-veterans-affairs/vets.gov-team/issues/1445

This PR accomplishes the following

- Encapsulate SAML health checks into its own class
- Add specs for SAML health checks
- Add a certificate validity check to IDP signing cert

### CODE REVIEW

File: lib/saml/settings_service.rb [hunk 1]
```diff
@@ -11,18 +12,21 @@ class << self
...
       def saml_settings
-        if @fetch_attempted.nil? || metadata_successfully_retrieved?
+        if HealthStatus.healthy?
           merged_saml_settings
         else
           refresh_saml_settings
         end
       end
 
+      # only made public for specs
```
Comment (kreek) [inline lib/saml/settings_service.rb]: Is this true? :) it's called from HealthStatus#metadata_received?

File: lib/saml/health_status.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        return StatusMessages::NOT_ATTEMPTED unless fetch_attempted?
+        return StatusMessages::MISSING       unless metadata_received?
+        return StatusMessages::CERT_INVALID  unless idp_cert_valid?
+        ''
+      end
+
+      private
+
+      def fetch_attempted?
+        SettingsService.fetch_attempted == true
```
Comment (omgitsbillryan) [inline lib/saml/health_status.rb]: `fetch_attempted` may be `nil`.  I want this method to return only `true` or `false`

File: lib/saml/health_status.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+      def healthy?
+        error_message.empty?
+      end
+
+      def error_message
+        return StatusMessages::NOT_ATTEMPTED unless fetch_attempted?
+        return StatusMessages::MISSING       unless metadata_received?
+        return StatusMessages::CERT_INVALID  unless idp_cert_valid?
+        ''
```
Comment (omgitsbillryan) [inline lib/saml/health_status.rb]: I added logic to `SettingsService.saml_settings` to log an error using HealthStatus.error_message`

**Comment by patrickvinograd** [comment]
Apologies for leaving this hanging. Looking now.

**Pull request merged**